File: troc-12/troc-12-bakteri/solution_rama.py
# Author   : rama_pang
# Problem  : troc-12-bakteri
# Time     : O(N M)
# Memory   : O(N M)

from sys import stdin, stdout

def main():
  N, M, K = [int(x) for x in stdin.readline().split()]
  ans = [['.' for j in range(M)] for i in range(N)]

  for i in range(N - 1, -1, -2):
    ans[i][0] = 'X'
  for i in range(M - 1, -1, -2):
    ans[0][i] = 'X'
  
  K -= sum(element.count('X') for element in ans)
  for i in range(N):
    for j in range(M):
      if ans[i][j] == '.' and K > 0:
        ans[i][j] = 'X'
        K -= 1
  
  if K < 0:
    stdout.write('TIDAK\n')
  else :
    stdout.write('YA\n')
    stdout.write('\n'.join([''.join(i) for i in ans]) + '\n')

if __name__ == '__main__':
  main()

File: troc-12/troc-12-bakteri/solution_rama.cpp
// Author   : rama_pang
// Problem  : troc-12-bakteri
// Time     : O(N M)
// Memory   : O(N M)

#include <bits/stdc++.h>
using namespace std;

int main() {
  int N, M, K;
  cin >> N >> M >> K;

  vector<string> A(N, string(M, '.'));

  for (int i = N - 1; i >= 0; i -= 2) {
    A[i][0] = 'X';
  }
  for (int i = M - 1; i >= 0; i -= 2) {
    A[0][i] = 'X';
  }

  for (int i = 0; i < N; i++) {
    K -= count(begin(A[i]), end(A[i]), 'X');
  }

  for (int i = 0; i < N; i++) {
    for (int j = 0; j < M; j++) {
      if (A[i][j] == '.' && K > 0) {
        A[i][j] = 'X';
        K--;
      }
    }
  }

  if (K < 0) {
    cout << "TIDAK\n";
    return 0;
  }

  cout << "YA\n";
  for (int i = 0; i < N; i++) {
    cout << A[i] << "\n";
  }
  return 0;
}
